web/src/settings_org.js

```diff
@@ -704,29 +708,193 @@ export function discard_property_element_changes(elem, for_realm_default_setting
     update_dependent_subsettings(property_name);
 }
 
+export function valid_to(deletes_in) {
+    const time_valid = Number.parseFloat(deletes_in);
+    if (time_valid === 0) {
+        return $t({defaultMessage: "Data will be deleted immediately"});
+    }
+
+    if (!time_valid) {
+        return $t({defaultMessage: "Data will not be automatically deleted"});
+    }
+
+    // The below is a duplicate of timerender.get_full_datetime, with a different base string.
+    const valid_to = add(new Date(), {minutes: time_valid});
+    const date = timerender.get_localized_date_or_time_for_format(valid_to, "dayofyear_year");
+    const time = timerender.get_localized_date_or_time_for_format(valid_to, "time");
+
+    return $t({defaultMessage: "Data to be deleted on {date} at {time}"}, {date, time});
+}
+
 export function deactivate_organization(e) {
     e.preventDefault();
     e.stopPropagation();
 
     function do_deactivate_realm() {
+        let deletes_in_days = $("#deletes_in").val();
+        // See settings_config.realm_deletion_in_values for why we do this conversion.
+        if (deletes_in_days === "null") {
+            deletes_in_days = JSON.stringify(null);
+        } else if (deletes_in_days === "custom") {
+            const deletes_in_minutes = Number.parseFloat(
+                settings_ui.get_custom_input_time_in_minutes(
+                    custom_deletion_time_unit,
+                    custom_deletion_time_input,
+                ),
+            );
+            deletes_in_days = deletes_in_minutes / (60 * 24);
```

Comment:
What happens if the custom time is not an integer number of days here -- does this round or send a floating point value?

Reply:
No, the value is not rounded. Instead floating number is sent to API and API is returning a json error `deletion_delay_days is not an integer`. I don't think we need to make any changes in this behaviour. Let me know if any thing needs to be changed.

Reply:
I think if we only want to allow the users to set value in days, then we should not provide other options in the UI. It does not make sense to give users the option of choosing time unit in minutes and then expect them to give the input such that it can be converted to integer number of days.

Reply:
We are not allowing users to give time in terms of minutes. User can only give time in terms of number of days or weeks. The question above was what If a user gives a value like 3.2 days.

Reply:
Ok. So we can prevent that in UI itself I think. We can just disable the submit button of modal in that case I guess.

Reply:
Yeah that seems reasonable; @Ujjawal3 have you done something like that, or is this item still open? I think it could reasonably be left as a follow-up issue because I don't think anyone is in practice going to enter `3.2` anyway.

Reply:
No, I have not done any work for it.